test/compilable/crlf_dshell.d

```diff
@@ -0,0 +1,52 @@
+module test;
+import dshell;
+int runTest()
+{
+    // Test CRLF and mixed line ending handling in D lexer.
+    const src = buildPath(RESULTS_TEST_DIR, TEST_NAME ~ ".d");
+    
+    src.createFile(
+          "#!/usr/bin/env dmd -run\r\n"
+        ~ "\r\n"
+        ~ "#line 4\r\n"
+        ~ "\r\n"
+        ~ "void main()\r\n"
+        ~ "{\r\n"
+        ~ "}\r\n"
+        ~ "\r\n"
+        ~ "// single-line comment\r\n"
+        ~ "\r\n"
+        ~ "/*\r\n"
+        ~ "  multi-line comment\r\n"
+        ~ "*/\r\n"
+        ~ "\r\n"
+        ~ "/+\r\n"
+        ~ "  nested comment\r\n"
+        ~ "+/\r\n"
+        ~ "\r\n"
+        ~ "/**\r\n"
+        ~ "  doc comment\r\n"
+         ~ "*/\r\n"
+        ~ "void documentee() {}\r\n");
+    
+    src.appendFile("// mixed\n// line\n// endings\n");
+    src.appendFile("void fun()\n{\n}\n");
+    
+    src.appendFile("enum str = \"\r\nfoo\r\nbar\nbaz\r\n\";\n");
+    src.appendFile(format("static assert(str == \"%s\");\n", `\nfoo\nbar\nbaz\n`));
+    
+    src.appendFile("enum bstr = `\r\nfoo\r\nbar\nbaz\r\n`;\n");
+    src.appendFile(format("static assert(bstr == \"%s\");\n", `\nfoo\nbar\nbaz\n`));
+    
+    src.appendFile("enum wstr = q\"EOF\r\nfoo\r\nbar\nbaz\r\nEOF\";\n");
+    src.appendFile(format("static assert(wstr == \"%s\");\n", `foo\nbar\nbaz\n`));
+    
+    src.appendFile("enum dstr = q\"(\r\nfoo\r\nbar\nbaz\r\n)\";\n");
+    src.appendFile(format("static assert(dstr == \"%s\");\n", `\nfoo\nbar\nbaz\n`));
+
+    const obj = buildPath(RESULTS_TEST_DIR, TEST_NAME ~ "a" ~ env.OBJ);
+    run([env.DMD, "-m"~env.MODEL, "-c", "-D", "-Dd" ~ RESULTS_TEST_DIR, "-of" ~ obj, src]);
```

Comment:
Why not use your `dmd` struct from below?

Reply:
wasn't sure which one was better....what do you think?